packages/neutron-sdk/src/bindings/oracle/query.rs

```diff
@@ -0,0 +1,39 @@
+use crate::bindings::oracle::types::{CurrencyPair, QuotePrice};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum OracleQuery {
+    GetAllCurrencyPairs {},
+    GetPrice { currency_pair: CurrencyPair },
+    GetPrices { currency_pair_ids: Vec<String> },
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct GetPriceResponse {
+    /// **price** represents the quote-price for the CurrencyPair given in
+    /// GetPriceRequest (possibly nil if no update has been made)
+    pub price: QuotePrice,
+    /// **nonce** represents the nonce for the CurrencyPair if it exists in state
+    pub nonce: u64,
+    /// **decimals* represents the number of decimals that the quote-price is
+    /// represented in. For Pairs where ETHEREUM is the quote this will be 18,
+    /// otherwise it will be 8.
+    pub decimals: u64,
+    /// *id** represents the identifier for the CurrencyPair.
+    pub id: Option<u64>,
```

Comment:
i think `serde(default)` is a good choice here, to avoid unnecessary `id.unwrap()`, in this particular case `0 == None`